packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -490,7 +492,12 @@ export class GarbageCollector implements IGarbageCollector {
             }
         }
 
+        // Sweep V0 is a test mode for using a very short timeout in order to enable testing of real sweep behavior.
+        // It's similar to testMode but rather than deleting immediately, it exercises the Sweep timers and codepaths
+        this.sweepV0TimeoutMs = this.mc.config.getNumber("Fluid.GarbageCollection.TestOverride.SweepV0TimeoutMs");
```

Comment:
I think the usage of this.sweepTimeoutMs across the code is wrong.
Sweep duration should be the same for all collaborators in the same document for system to work correctly. Even hard coding the value in the code does not provide that guarantee (it may change in next version), using feature gates make it even worse. I think the best course of action for this value (from code or feature gate) to be used to be written into file metadata, and every client using same value from metadata for lifetime of the file. Allowing to increase this value (in next summaries), but never decrease.

It's not ideal (test files are "contaminated" with small timeout forever), but I think we can work with it (whoever creates such files should put a big warning at the top of the file that it's for test purposes - it required anyway, as uneducated user opening this file can run into troubles).

Reply:
This reminds me, we should have some way of being able to ensure this value is consistent across clients. It feels like this needs to be an op of some sorts. A half-baked thought would be to create a GC datastore. We would be using our own datastore logic, and we could definitely take advantage of a significant portion of the logic used to sync ops/datastores. It would help us solve the problem of setting sessionExpiry and modifying it. It would help us solve the problem of having inconsistent sweep timings. It feels like the GC datastore would be a special datastore. Granted I do not know what the cost of doing something like this is.

Pros:
- resuses a lot of logic to ensure consistency across clients without having to rewrite everything
- allows us to take advantage of any sort of performance improvements to datastores and the DDS we use
- potentially more recovery abilities, might be able to leverage rollback capabilities from certain DDSs, so we have the potential to rollback without having to design it ourselves as a GC team.
- provides an example to others of how to use Datastores (may not be a con if we don't want others to do this)
- may allow us to remove a bunch of GC specific summary things.

Cons:
- One issue is that garbage collection would be creating its own ops. 
- The cost of doing this vs. slowly solving those problems over time.
- We may be exposing functionality that others may use to write their own GC algorithms
- All clients would have to process the GC ops.

Reply:
Thanks for pointing this out, both of you. My next step in this PR is to address this.  See also the doc I shared about adjusting "Sweep V0" plan to "Hyper Sweep Mode" where all relevant timers are shortened together to keep the overall flow consistent.  Planning to implement that here, but gotta figure out about keep these timers locked per file first.

Reply:
Ok in latest iteration if we use a TestMode property in the metadata to stamp the file.